pyvista/plotting/charts.py

```diff
@@ -0,0 +1,3375 @@
+            f_h = t_h / self._fig.dpi
+            self._fig.set_size_inches(f_w, f_h)
+            self.position = (self._loc[0]*r_w, self._loc[1]*r_h)
+
+    def redraw(self):
+        """Redraw the chart after manually updating the matplotlib figure."""
+        # TODO: not sure whether this is still needed, as calling plotter.update() seems to do the trick already.
+        # Manual call, so make sure canvas is redrawn first (which will callback to _redraw with a proper event defined)
+        self._canvas.draw()
+
+    def _redraw(self, event):
+        # Called from draw_event callback
+        img = np.frombuffer(self._canvas.buffer_rgba(), dtype=np.uint8)  # Store figure data in numpy array
+        w, h = self._canvas.get_width_height()
+        img_arr = img.reshape([h, w, 4])
+        img_data = pyvista.Texture(img_arr).to_image()  # Convert to vtkImageData
+        self.SetImage(img_data)
+
+    def _render_event(self, *args, **kwargs):
+        self._resize()  # Update figure dimensions if needed
+        self.redraw()  # Redraw figure
+
+    @property
+    def _geometry(self):
+        r_w, r_h = self._renderer.GetSize()
+        t_w = self._size[0]*r_w
+        t_h = self._size[1]*r_h
+        return (*self.position, t_w, t_h)
+
+    @_geometry.setter
+    def _geometry(self, value):
+        raise AttributeError(f'Cannot set the geometry of {type(self).__class__}')
+
+    # Below code can be used to customize the chart's background without a _ChartBackground instance
+    # @property
+    # def background_color(self):
+    #     return self._bg_color
+    #
+    # @background_color.setter
+    # def background_color(self, val):
+    #     color = parse_color(val) if val is not None else [1, 1, 1, 1]
+    #     opacity = color[3] if len(color) == 4 else 1
+    #     self._bg_color = color
+    #     self._fig.patch.set_color(color[:3])
+    #     self._fig.patch.set_alpha(opacity)
+    #     for ax in self._fig.axes:
+    #         ax.patch.set_alpha(0 if opacity < 1 else 1)  # Make axes fully transparent if opacity is lower than 1
+
+    @property
+    def position(self):
+        """Chart position w.r.t the bottom left corner (in pixels)."""
+        return self.GetPosition()
+
+    @position.setter
+    def position(self, val):
+        assert len(val) == 2
+        self.SetPosition(*val)
+
+
+class Charts:
+    """Collection of charts for a renderer."""
+
+    def __init__(self, renderer):
+        """Create a new collection of charts for the given renderer.
+
+        Users should typically not directly create new instances of this class, but use the dedicated
+        ``Plotter.add_chart`` method.
+        """
+        self._charts = []
+
+        # Postpone creation of scene and actor objects until they are
+        # needed.
+        self._scene = None
+        self._actor = None
+        self._renderer = renderer
+
+    def _setup_scene(self):
+        """Setup a new context scene and actor for these charts."""
```

Comment:
I think the style error here might ba false positive. See if changing to "Set up..." helps:
```suggestion
        """Set up a new context scene and actor for these charts."""
```
I suspect the linter thinks "Setup" is a noun here.